### PR TITLE
Issue #64: removed percent encoding on slugs and parameter values

### LINKED ISSUE #64: Parameters should be percent escaped
In Vapor 2 the following code:
```swift
get("hello", ":id") { req in
            let id = req.parameters["id"]?.string

            var json = JSON()
            try json.set("hello", id)
            return json
        }
```

Would produce the following output:

- GET /hello/abc1234 `{"hello":"abc1234"}`
- GET /hello/abc|1234 `{"hello":"abc|1234"}`
- GET /hello/abc%7C1234 `{"hello":"abc|1234"}`

In Vapor 3 instead:

- GET /hello/abc1234 `{"hello":"abc1234"}`
- GET /hello/abc|1234 `{"hello":"abc%7C1234"}` // Notice that this is automatically encoded by any client before sending
- GET /hello/abc%7C1234 `{"hello":"abc%7C1234"}`

This to me seems like a regression.

The code that was taking care of this can be found on `Router.swift` on tag `2.2.1`.
```swift
public func slugs(for path: [String]) -> Parameters {
        var slugs: [String: Parameters] = [:]
        slugIndexes.forEach { key, index in
            guard let val = path[safe: index]
                .flatMap({ $0.removingPercentEncoding })
                .flatMap({ Parameters.string($0) })
                else { return }

            if let existing = slugs[key] {
                var array = existing.array ?? [existing]
                array.append(val)
                slugs[key] = .array(array)
            } else {
                slugs[key] = val
            }
        }
        return .object(slugs)
    }
```

### PR DESCRIPTION
**Pull request by valeriomazzeo**
Fixes #64

### CODE REVIEW

**Comment by valeriomazzeo** [comment]
CI error doesn't seem to be related to these code changes:
```
/root/vapor: error: package at '/root/vapor' requires a minimum Swift tools version of 4.2.0 (currently 4.1.0)
Exited with code 1
```

File: Sources/Routing/Routing/TrieRouter.swift [hunk 1]
```diff
@@ -85,8 +85,8 @@ public final class TrieRouter<Output> {
                 // if no constant routes were found that match the path, but
                 // a dynamic parameter child was found, we can use it
                 let value = ParameterValue(
-                    slug: String(data: parameter.value, encoding: .utf8) ?? "",
-                    value: path.routerParameterValue
+                    slug: String(data: parameter.value, encoding: .utf8)?.removingPercentEncoding ?? "",
```
Comment (tanner0101) [inline Sources/Routing/Routing/TrieRouter.swift]: The slug comes from the route registration (i.e, `router.get("users/:id")`). We shouldn't need to percent decode it.

**Comment by tanner0101** [comment]
Regarding the CI error, we just need to switch this line to cloning Vapor's `3` branch instead of master (which is now Vapor 4): https://github.com/vapor/routing/blob/3/circle.yml#L34

**Comment by penny-coin** [comment]
Hey @valeriomazzeo, you just merged a pull request, have a coin! 

You now have 2 coins.

**Pull request merged**